### PR TITLE
HPCC-17087 Fix remote file appendFile functionality.

### PR DESCRIPTION
**Pull request by jakesmith**
Previously appending a file with no length (meaning whole file)
would error due to misinterpreting the length.

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [x] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:

This was tested via the unittests added in JIRA : HPCC-17089

### CODE REVIEW

**Comment by jakesmith** [comment]
@mckellyln - please review

File: common/remote/sockfile.cpp [hunk 1]
```diff
@@ -4283,7 +4283,6 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
         file->rename(toname);
         reply.append((unsigned)RFEnoerror);
         return true;
-        return false;
```
Comment (jakesmith) [inline common/remote/sockfile.cpp]: NB: incidental change/redundant code.

File: common/remote/sockfile.cpp [hunk 2]
```diff
@@ -4299,7 +4298,6 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
         file->move(toname);
         reply.append((unsigned)RFEnoerror);
         return true;
-        return false;
```
Comment (jakesmith) [inline common/remote/sockfile.cpp]: NB: incidental change/redundant code.

File: common/remote/sockfile.cpp [hunk 3]
```diff
@@ -2453,7 +2453,7 @@ class CRemoteFileIO : implements IFileIO, public CInterface
         offset_t ret;
         replyBuffer.read(ret);
 
-        if ((ret==(offset_t)-1) || (ret < len))
+        if ((ret==(offset_t)-1) || ((len != ((offset_t)-1)) && (ret < len)))
```
Comment (mckellyln) [inline common/remote/sockfile.cpp]: ok, so if len of -1 was passed into appendFile() that means append the entire file, right ?

File: common/remote/sockfile.cpp (hunk 1 above)
Comment (mckellyln) [inline common/remote/sockfile.cpp]: ok, so return value doesn't really matter but just being consistent ?

Comment (jakesmith) [inline common/remote/sockfile.cpp]: It does matter..

Any remote rename whose result was checked would have seen a false negative.

File: common/remote/sockfile.cpp (hunk 3 above)
Comment (jakesmith) [inline common/remote/sockfile.cpp]: Yep.

**Comment by mckellyln** [comment]
Code looks fine to merge.

**Review by mckellyln — APPROVED** [review]
Code approved

File: common/remote/sockfile.cpp (hunk 1 above)
Comment (richardkchapman) [inline common/remote/sockfile.cpp]: Deleting unreachable code...

File: common/remote/sockfile.cpp (hunk 2 above)
Comment (richardkchapman) [inline common/remote/sockfile.cpp]: Will also resolve a coverity error I suspect...

**Comment by richardkchapman** [comment]
@jakesmith You say you added tests to cover - they are under a separate ticket?

**Comment by jakesmith** [comment]
I thought I commented/referenced the JIRA with the unit tests, seems not, have updated PR descriptor Testing section now to reference HPCC-17089 with unittests.

**Pull request merged**